internal/icinga2/client_api.go

```diff
@@ -427,26 +438,56 @@ func (client *Client) listenEventStream() error {
 		)
 		switch respT := resp.(type) {
 		case *StateChange:
-			// Only process HARD states
+			// Ignore any state changes while in downtime
+			if respT.DowntimeDepth > 0 {
+				stateType := "HARD"
+				if respT.StateType == StateTypeSoft {
+					stateType = "SOFT"
+				}
+
+				client.Logger.Debugf("Skipping %q state change while in downtime, %#v", stateType, respT)
+				continue
+			}
+
+			// Only process HARD states.
+			// The checkable might also be in the flapping state, but Icinga 2 doesn't expose "is_flapping" to the
+			// event streams for state change. So, the only thing we can do here is to ignore soft state changes.
 			if respT.StateType == StateTypeSoft {
 				client.Logger.Debugf("Skipping SOFT State Change, %#v", respT)
 				continue
 			}
 
 			ev, err = client.buildHostServiceEvent(client.Ctx, respT.CheckResult, respT.State, respT.Host, respT.Service)
 			evTime = respT.Timestamp.Time()
-
 		case *AcknowledgementSet:
-			ev, err = client.buildAcknowledgementEvent(client.Ctx, respT.Host, respT.Service, respT.Author, respT.Comment)
+			ev, err = client.buildAcknowledgementEvent(client.Ctx, respT.Host, respT.Service, respT.Author, respT.Comment, false)
+			evTime = respT.Timestamp.Time()
+		case *AcknowledgementCleared:
+			ev, err = client.buildAcknowledgementEvent(client.Ctx, respT.Host, respT.Service, "", "", true)
 			evTime = respT.Timestamp.Time()
-
-		// case *AcknowledgementCleared:
 		// case *CommentAdded:
 		// case *CommentRemoved:
 		// case *DowntimeAdded:
-		// case *DowntimeRemoved:
-		// case *DowntimeStarted:
-		// case *DowntimeTriggered:
+		case *DowntimeRemoved:
+			ev, err = client.buildDowntimeEvent(client.Ctx, respT.Downtime, false)
+			evTime = respT.Timestamp.Time()
+		case *DowntimeStarted:
+			if !respT.Downtime.IsFixed {
+				continue
+			}
+
+			ev, err = client.buildDowntimeEvent(client.Ctx, respT.Downtime, true)
+			evTime = respT.Timestamp.Time()
+		case *DowntimeTriggered:
+			if respT.Downtime.IsFixed {
+				continue
+			}
```

Comment:
A comment on why these two `continue` are here would be highly appreciated, especially given that it's confusing and the reason is outside of this repository.

Reply:
Did you actually read the comments on that field?

Reply:
Yes, but indeed not while looking over this file.

I'm be looking for something like "Match the behavior of Icinga 2 which notifies for fixed downtimes only in OnDowntimeStarted: https://github.com/Icinga/icinga2/blob/v2.14.2/lib/icinga/checkable.cpp#L39", that's also missing there.